File: internal/dto/analytics.go
package dto

type AnalyticsResponse struct {
	IncomeThisMonth  string `json:"income_this_month"`
	ExpenseThisMonth string `json:"expense_this_month"`
	CreditLoad       string `json:"credit_load"`
}

type OperationStatisticsResponse struct {
	EntityType     string                     `json:"entity_type"`
	EntityID       int64                      `json:"entity_id"`
	Currency       string                     `json:"currency"`
	OperationCount int64                      `json:"operation_count"`
	IncomeCount    int64                      `json:"income_count"`
	ExpenseCount   int64                      `json:"expense_count"`
	TotalIncome    string                     `json:"total_income"`
	TotalExpense   string                     `json:"total_expense"`
	NetAmount      string                     `json:"net_amount"`
	ByType         []OperationTypeResponse    `json:"by_type"`
	ByStatus       []OperationStatusResponse  `json:"by_status"`
	Operations     []OperationHistoryResponse `json:"operations"`
}

type OperationTypeResponse struct {
	Type         string `json:"type"`
	Count        int64  `json:"count"`
	TotalIncome  string `json:"total_income"`
	TotalExpense string `json:"total_expense"`
	NetAmount    string `json:"net_amount"`
}

type OperationStatusResponse struct {
	Status      string `json:"status"`
	Count       int64  `json:"count"`
	TotalAmount string `json:"total_amount"`
}

type OperationHistoryResponse struct {
	ID            int64  `json:"id"`
	Direction     string `json:"direction"`
	Type          string `json:"type"`
	Status        string `json:"status"`
	Amount        string `json:"amount"`
	Currency      string `json:"currency"`
	Description   string `json:"description,omitempty"`
	FromAccountID *int64 `json:"from_account_id,omitempty"`
	ToAccountID   *int64 `json:"to_account_id,omitempty"`
	FromCardID    *int64 `json:"from_card_id,omitempty"`
	ToCardID      *int64 `json:"to_card_id,omitempty"`
	CreatedAt     string `json:"created_at"`
}

File: web/src/types/analytics.ts
export type AnalyticsResponse = {
  income_this_month: string
  expense_this_month: string
  credit_load: string
  active_credits_count: number
}

export type OperationDirection = 'income' | 'expense' | 'neutral'

export type OperationTypeStatistics = {
  type: string
  count: number
  total_income: string
  total_expense: string
  net_amount: string
}

export type OperationStatusStatistics = {
  status: string
  count: number
  total_amount: string
}

export type OperationHistoryItem = {
  id: number
  direction: OperationDirection
  type: string
  status: string
  amount: string
  currency: string
  description?: string
  from_account_id?: number
  to_account_id?: number
  from_card_id?: number
  to_card_id?: number
  created_at: string
}

export type OperationStatisticsResponse = {
  entity_type: 'account' | 'card'
  entity_id: number
  currency: string
  operation_count: number
  income_count: number
  expense_count: number
  total_income: string
  total_expense: string
  net_amount: string
  by_type: OperationTypeStatistics[]
  by_status: OperationStatusStatistics[]
  operations: OperationHistoryItem[]
}
